### DISCUSSION TITLE
Client will set column to "" even if 'Allow empty' is unchecked

### DISCUSSION THREAD

**Discussion by nathanlindsay** [supabase/supabase#980] [Questions] [answered]
If you uncheck 'Allow empty' for a column, the client is still able to set the value of that column to "" without throwing any errors.

Is this intended behaviour?

**Comment by steve-chavez** [comment, marked as answer]
Allow empty is for null columns, it doesn't prevent empty `text` fields. For doing this, you could add a check constraint:

```sql
alter table tbl add check (trim(col) <> '')
```

(Maybe the GUI can include a shortcut for the check)
